ugs-core/src/com/willwinder/universalgcodesender/SmoothieCommunicator.java

```diff
@@ -19,9 +19,33 @@ This file is part of Universal Gcode Sender (UGS).
 package com.willwinder.universalgcodesender;
 
 /**
- *
  * @author wwinder
  */
-public class SmoothieCommunicator extends GrblCommunicator {
+public class SmoothieCommunicator extends BufferedCommunicator {
 
+    @Override
+    public int getBufferSize() {
+        return 255;
+    }
+
+    @Override
+    protected void sendingCommand(String command) {
+
+    }
+
+    @Override
+    protected boolean processedCommand(String response) {
+        return SmoothieUtils.isOkErrorAlarmResponse(response) || SmoothieUtils.isVersionResponse(response);
+    }
+
+    @Override
+    protected boolean processedCommandIsError(String response) {
+        return GrblUtils.isErrorResponse(response);
+    }
+
+    @Override
+    public boolean allowMoreCommands() {
+        // Force single step
+        return this.activeCommandList.isEmpty();
```

Comment:
You should be able to override `this.singleStepModeEnabled` (or override the setter to ignore anything that disables it) instead of making `activeCommandList` protected.